pkg/vault/test/foundry/fork/YieldBearingPoolSwapBase.t.sol

```diff
@@ -512,10 +380,20 @@ abstract contract YieldBearingPoolSwapBase is BaseVaultTest {
         // order of 1e12 (1e18/1e6) in amountOut of the last buffer.
         // But, if it's the opposite case, 1e6/1e18 is rounded to 0, but the max error is actually 1 (the error in the
         // tokenOut token itself), so the division is incremented by 1.
-        // Finally, the error can be slightly bigger than the division of the factors, since each token has a rate. The
-        // error is multiplied by 2 to give some space for rounding errors related to the rates.
-        uint256 absTolerance = (2 *
-            (paths[0].tokenIn == _token1Fork ? _token2Factor / _token1Factor : _token1Factor / _token2Factor)) + 1;
+        uint256 decimalError = (
+            paths[0].tokenIn == _token1Fork ? _token2Factor / _token1Factor : _token1Factor / _token2Factor
+        ) + 1;
+
+        // Query and actual operation can return different results, depending on the difference of decimals. The error
+        // is amplified by the rate of the token out and by `vaultConvertFactor` when using the buffer liquidity in
+        // the actual operation.
+        uint256 sharesError = vars.ybTokenOut.convertToAssets(
+            (withBufferLiquidity ? vaultConvertFactor : 1) * decimalError
+        );
+        uint256 absTolerance = sharesError + (withBufferLiquidity ? vaultConvertFactor : 1);
+        // If convertToAssets return 0, absTolerance may be smaller than the error introduced by the difference of
+        // decimals, so keep the decimalError.
+        absTolerance = absTolerance > decimalError ? absTolerance : decimalError;
```

Comment:
All of this is just to get an upper bound for the absolute error between the query and the actual operation right?
I'd clarify that in the comments: "The error _between the query and the actual operation_ is proportional..."

Reply:
I'm happy knowing that the relative error is bounded by 0.01%.
Other than that, with the conversion factor is the error always one-sided? i.e. is the query always worse than the actual operation for the user then? That would be a good thing to verify: besides using `assertApproxEqAbs` or `assertApproxEqRel` we can use `gt` / `lt` (not saying we should remove the existing assertions, maybe add one more if it's easy for documentation purposes as well).

Reply:
Done!

Reply:
I couldn't assert that queries are always greater or smaller than the actual operation. Not related to the vaultConvertFactor, maybe related to a rate calculation that introduces some uncertainty. I can check later why this is not the case.